src/SpectatorView.Native/SpectatorView.Compositor/Compositor/CompositorInterface.cpp

```diff
@@ -159,7 +159,7 @@ bool CompositorInterface::InitializeVideoEncoder(ID3D11Device* device)
     return videoEncoder1080p->Initialize(device) && videoEncoder4K->Initialize(device);
 }
 
-void CompositorInterface::StartRecording(VideoRecordingFrameLayout frameLayout)
+void CompositorInterface::StartRecording(VideoRecordingFrameLayout frameLayout, LPWSTR lpFilePath, int lpFilePathLength)
```

Comment:
I know this would be a larger change, but I'd prefer a design where the caller to StartRecording can specify the target path. This would mean that the logic in FindUniqueFileName would need to be written in C# in the compositor instead of here, but it would make this library more flexible in that you could choose to record wherever you want without rebuilding the native compositor.